x-pack/platform/packages/shared/onechat/onechat-common/chat/events.ts

```diff
@@ -89,6 +91,21 @@ export const isToolResultEvent = (event: OnechatEvent<string, any>): event is To
   return event.type === ChatEventType.toolResult;
 };
 
+// Prompt request
+
+export interface PromptRequestEventData {
+  prompt: PromptRequest;
+  source: PromptRequestSource;
+}
+
+export type PromptRequestEvent = ChatEventBase<ChatEventType.promptRequest, PromptRequestEventData>;
```

Comment:
Technically the execute ends when a prompt request is received, so I don't this the event is going to be very useful (the UI can just use the final round event to show the confirmation), but I still added it for consistency